src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BasePatternSwitchLocalRewriter.cs

```diff
@@ -0,0 +1,415 @@
+            /// it may occur when something (from the enclosing expression) is on the stack.
+            /// To satisfy the verifier, the caller must arrange forward jumps to these labels. The set of
+            /// states whose labels will need such forward jumps (if something can be on the stack) is stored in
+            /// _backwardLabels. In practice, this is exclusively the set of states that are reached
+            /// when a when-clause evaluates to false.
+            /// PROTOTYPE(patterns2): This is a placeholder. It is not used yet for lowering the
+            /// switch expression, where it will be needed.
+            /// </summary>
+            private readonly ArrayBuilder<BoundDecisionDag> _backwardLabels = ArrayBuilder<BoundDecisionDag>.GetInstance();
+
+            /// <summary>
+            /// The lowered decision dag. This includes all of the code to decide which pattern
+            /// is matched, but not the code to assign to pattern variables and evaluate when clauses.
+            /// </summary>
+            protected readonly ArrayBuilder<BoundStatement> _loweredDecisionDag = ArrayBuilder<BoundStatement>.GetInstance();
+
+            /// <summary>
+            /// The label in the code for the beginning of code for each node of the dag.
+            /// </summary>
+            private readonly Dictionary<BoundDecisionDag, LabelSymbol> _dagNodeLabels = new Dictionary<BoundDecisionDag, LabelSymbol>();
+
+            protected BasePatternSwitchLocalRewriter(LocalRewriter localRewriter, BoundExpression loweredInput, ImmutableArray<SyntaxNode> arms, BoundDecisionDag decisionDag)
+                : base(localRewriter, loweredInput)
+            {
+                foreach (var arm in arms)
+                {
+                    _switchArms.Add(arm, new ArrayBuilder<BoundStatement>());
+                }
+
+                ImmutableArray<BoundDecisionDag> sortedNodes = decisionDag.TopologicallySortedNodes();
+                ComputeLabelSet(sortedNodes);
+                LowerDecisionDag(sortedNodes);
+            }
+
+            private void ComputeLabelSet(ImmutableArray<BoundDecisionDag> sortedNodes)
+            {
+                // Nodes with more than one predecessor are assigned a label
+                var hasPredecessor = PooledHashSet<BoundDecisionDag>.GetInstance();
+                void notePredecessor(BoundDecisionDag successor)
+                {
+                    if (successor != null && !hasPredecessor.Add(successor))
+                    {
+                        GetDagNodeLabel(successor);
+                    }
+                }
+
+                foreach (BoundDecisionDag node in sortedNodes)
+                {
+                    switch (node)
+                    {
+                        case BoundWhenClause w:
+                            GetDagNodeLabel(node);
+                            //Debug.Assert(w.WhenTrue != null);
+                            //GetDagNodeLabel(w.WhenTrue);
```

Comment:
Remove?

Reply:
Recorded in [https://github.com/dotnet/roslyn/issues/25663](https://github.com/dotnet/roslyn/issues/25663)

---
In reply to: [176260962](https://github.com/dotnet/roslyn/pull/25544#discussion_r176260962) [](ancestors = 176260962)